Let TensorPipe detect peer access

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #50788 Add CUDA RPC stress tests
* **#50676 Let TensorPipe detect peer access**

Differential Revision: [D25941962](https://our.internmc.facebook.com/intern/diff/D25941962)

mrshenli committed:
Let TensorPipe detect peer access

[ghstack-poisoned]

beauby commented on torch/testing/_internal/distributed/rpc/rpc_test.py:
```diff
@@ -25,7 +25,6 @@
 )
 from torch.testing._internal.common_distributed import (
     skip_if_lt_x_gpu,
-    skip_if_no_peer_access,
```
Was the `skip_if_no_peer_access` decorator added specifically for these tests? If so, we might want to remove it altogether.

mrshenli replied:
yep, let me remove that function

lw commented:
It seems some occurrence was missed?
```
Jan 18 06:43:15 Traceback (most recent call last):
Jan 18 06:43:15   File "distributed/rpc/test_process_group_agent.py", line 7, in <module>
Jan 18 06:43:15     from torch.testing._internal.distributed.rpc_utils import (
Jan 18 06:43:15   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/distributed/rpc_utils.py", line 40, in <module>
Jan 18 06:43:15     from torch.testing._internal.distributed.rpc.rpc_test import (
Jan 18 06:43:15   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/distributed/rpc/rpc_test.py", line 4461, in <module>
Jan 18 06:43:15     class TensorPipeAgentRpcTest(RpcAgentTestFixture):
Jan 18 06:43:15   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/distributed/rpc/rpc_test.py", line 4800, in TensorPipeAgentRpcTest
Jan 18 06:43:15     @skip_if_no_peer_access
Jan 18 06:43:15 NameError: name 'skip_if_no_peer_access' is not defined
```

lw commented:
Note: to make this stack "complete" you could also register the CUDA_GDR channel (although it won't have an impact on tests, since CircleCI doesn't have InfiniBand).

mrshenli committed:
Update on "Let TensorPipe detect peer access"


Differential Revision: [D25941962](https://our.internmc.facebook.com/intern/diff/D25941962)

[ghstack-poisoned]

mrshenli committed:
Update on "Let TensorPipe detect peer access"


Differential Revision: [D25941962](https://our.internmc.facebook.com/intern/diff/D25941962)

[ghstack-poisoned]

mrshenli committed:
Update on "Let TensorPipe detect peer access"


Differential Revision: [D25941962](https://our.internmc.facebook.com/intern/diff/D25941962)

[ghstack-poisoned]

mrshenli committed:
Update on "Let TensorPipe detect peer access"


Differential Revision: [D25941962](https://our.internmc.facebook.com/intern/diff/D25941962)

[ghstack-poisoned]

mrshenli committed:
Update on "Let TensorPipe detect peer access"


Differential Revision: [D25941962](https://our.internmc.facebook.com/intern/diff/D25941962)

[ghstack-poisoned]
